src/Program.cs

```diff
@@ -249,6 +258,68 @@ private static void InstallBase64Certificate(string base64, string password, Sto
             }
 
             AddToStore(cert, storeName, storeLocation);
+        }
+
+        private static void FindByThumbprint(string thumbprint, StoreName storeName, StoreLocation storeLocation)
+        {
+            var store = new X509Store(storeName, storeLocation);
+            store.Open(OpenFlags.ReadOnly);
+
+            if (store.Certificates.Count == 0)
+            {
+                return;
+            }
+
+            Console.WriteLine(
+                $"Searching for certificate thumbprint '{thumbprint}' in '{storeName}/{storeLocation}'...");
+            Console.WriteLine();
+
+            bool isFound = false;
+            foreach (var certificate in store.Certificates)
+            {
+                if (certificate.Thumbprint == null ||
+                    string.Compare(certificate.Thumbprint, thumbprint,
+                        StringComparison.OrdinalIgnoreCase) != 0)
+                {
+                    continue;
+                }
+
+                isFound = true;
+                var certificateInfo = new Dictionary<string, string>
+                {
+                    {"Subject", certificate.Subject},
+                    {"Issuer", certificate.Issuer},
+                    {"Serial Number", certificate.GetSerialNumberString()},
+                    {"Not Before", certificate.GetEffectiveDateString()},
+                    {"Not After", certificate.GetExpirationDateString()},
+                    {"Thumbprint", certificate.Thumbprint},
+                    {
+                        "Signature Algorithm",
+                        $"{certificate.SignatureAlgorithm.FriendlyName} ({certificate.SignatureAlgorithm.Value})"
+                    },
+                    {
+                        "PublicKey Algorithm",
+                        $"{certificate.PublicKey.Oid.FriendlyName} ({certificate.PublicKey.Oid.Value})"
+                    },
+                    {"Has PrivateKey", certificate.HasPrivateKey ? "Yes" : "No"}
+                };
```

Comment:
Duplicated code from `ListCertificates`, it's better to extract this code into a method.